it/uk/gov/hmrc/excisemovementcontrolsystemapi/PreValidateTraderControllerItSpec.scala

```diff
@@ -0,0 +1,183 @@
+
+
+class PreValidateTraderControllerItSpec extends PlaySpec
+  with GuiceOneServerPerSuite
+  with AuthTestSupport
+  with WireMockServerSpec
+  with BeforeAndAfterAll {
+
+  private lazy val wsClient: WSClient = app.injector.instanceOf[WSClient]
+  private val url = s"http://localhost:$port/traders/pre-validate"
+  private val eisUrl = "/emcs/pre-validate-trader/v1"
+  private val authErn = "GBWK002281023"
+
+  private val request = Json.toJson(getPreValidateTraderRequest)
+
+  override lazy val app: Application = {
+    wireMock.start()
+    WireMock.configureFor(wireHost, wireMock.port())
+    GuiceApplicationBuilder()
+      .configure(configureServer)
+      .overrides(
+        bind[AuthConnector].to(authConnector),
+      )
+      .build()
+  }
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    wireMock.resetAll()
+  }
+
+  override def afterAll(): Unit = {
+    super.afterAll()
+    wireMock.stop()
+  }
+
+  "Draft Excise Movement" should {
+
+    "return 200" in {
+      withAuthorizedTrader(authErn)
+      stubEISSuccessfulRequest()
+
+      val result = postRequest(request)
+      val expectedResult = getPreValidateTraderSuccessResponse.exciseTraderValidationResponse
+
+      result.status mustBe OK
+
+      withClue("return the json response") {
+        val responseBody = Json.parse(result.body).as[PreValidateTraderResponse].exciseTraderValidationResponse
+        responseBody.validationTimeStamp mustBe expectedResult.validationTimeStamp
+
+        responseBody.exciseTraderResponse(0) mustBe expectedResult.exciseTraderResponse(0)
+      }
+
+    }
+
+    "return not found if EIS returns not found" in {
+      withAuthorizedTrader(authErn)
+      stubEISErrorResponse(NOT_FOUND)
+
+      val result = postRequest(request)
+
+      result.status mustBe NOT_FOUND
+
+      withClue("return the EIS error response") {
+        result.body mustBe ""
+      }
+    }
+
+    "return bad request if EIS returns BAD_REQUEST" in {
+      withAuthorizedTrader(authErn)
+      stubEISErrorResponse(BAD_REQUEST)
+
+      postRequest(request).status mustBe BAD_REQUEST
+    }
+
+    "return 500 if EIS returns 500" in {
+      withAuthorizedTrader(authErn)
+      stubEISErrorResponse(INTERNAL_SERVER_ERROR)
+
+      postRequest(request).status mustBe INTERNAL_SERVER_ERROR
+    }
+
+    "return a Unauthorized (401) when no authorized trader" in {
+      withUnauthorizedTrader(InternalError("A general auth failure"))
+
+      postRequest(request).status mustBe UNAUTHORIZED
+    }
+
+    "return bad request (400) when json cannot be parsed" in {
+      withAuthorizedTrader(authErn)
+
+      postRequest(Json.toJson("{'blah': 'badJson'}")).status mustBe BAD_REQUEST
+    }
+
+    "return Unsupported Media Type (415)" in {
+      withAuthorizedTrader(authErn)
+      postRequest(Json.toJson(""), contentType = """application/xml""").status mustBe UNSUPPORTED_MEDIA_TYPE
+    }
+
+    "return bad request (400) when body is not xml" in {
```

Comment:
The body should be JSON, this test shouldn't be needed, but more worryingly, why does it pass?